tests/lava/magma/core/model/py/test_ports.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+import unittest
+import time
+from multiprocessing.managers import SharedMemoryManager
+import numpy as np
+
+from lava.magma.compiler.channels.interfaces import AbstractCspSendPort, \
+    AbstractCspRecvPort
+
+from lava.magma.core.model.py.ports import PyInPort, PyInPortVectorDense, \
+    PyOutPort, PyOutPortVectorDense
+
+from lava.magma.compiler.channels.pypychannel import PyPyChannel
+
+
+class MockInterface:
+    def __init__(self, smm):
+        self.smm = smm
+
+
+def get_channel(smm, data, size, name="test_channel") -> PyPyChannel:
+    mock = MockInterface(smm)
+    return PyPyChannel(
+        message_infrastructure=mock,
+        src_name=name,
+        dst_name=name,
+        shape=data.shape,
+        dtype=data.dtype,
+        size=size
+    )
+
+
+class TestPyPorts(unittest.TestCase):
+    def test_py_in_port_vector_dense_probe_single_csp_recv_port(self):
+        """Tests PyInPortVectorDense probe method for single recv_csp_port"""
+        smm = SharedMemoryManager()
+        try:
+            smm.start()
+
+            data = np.random.randint(5, size=(2, 4))
+
+            channel = get_channel(smm, data, data.size)
+            send_csp_port: AbstractCspSendPort = channel.src_port
+            recv_csp_port: AbstractCspRecvPort = channel.dst_port
+
+            send_py_port: PyOutPort = \
+                PyOutPortVectorDense([send_csp_port], None)
+            recv_py_port: PyInPort = \
+                PyInPortVectorDense([recv_csp_port], None)
+
+            recv_py_port.start()
+            send_py_port.start()
+
+            # Send data through PyOutPort
+            send_py_port.send(data)
+            # Sleep to let message reach the PyInPort
+            time.sleep(0.001)
+            # Probe PyInPort
+            probe_value = recv_py_port.probe()
+
+            # probe_value should be True if message reached the PyInPort
+            self.assertTrue(probe_value)
+
+            # Get data that reached PyInPort to empty buffer
+            _ = recv_py_port.recv()
+            # Probe PyInPort
+            probe_value = recv_py_port.probe()
+
+            # probe_value should be False since PyInPort's buffer was emptied
+            self.assertFalse(probe_value)
+        finally:
+            smm.shutdown()
+
+    def test_py_in_port_vector_dense_probe_multi_csp_recv_port(self):
+        """Tests PyInPortVectorDense probe method for multiple recv_csp_port"""
+        smm = SharedMemoryManager()
+        try:
+            smm.start()
+
+            data = np.random.randint(5, size=(2, 4))
```

Comment:
See above

Reply:
This has been addressed in latest commit.